### DISCUSSION TITLE
reading OSC messages sent by max for live via UDP.

### DISCUSSION THREAD

**Discussion by raul-antonio-estrada** [orchetect/swift-osc#38] [Q&A]
hi everyone, I am trying to get a simple swiftUI app going to fetch the messages that are being sent via UDP from max for live. 

I am a noob so I tried doing it without a library , so it was printing the value in an odd format like  "Received message: float,f?@O"  in a swiftUI app I created, that simply is supposed to print the message to console as a raw float. 

so I believe I can use this library to help me print the correct values to the console, which is supposed to be a float. 

the console was displaying this before I tried using the library if this helps 

"Received message: float,f?@O" 
clearly, not in the right format, in max for live it prints it as a float. 

I hope my question makes sense!

**Comment by mk-conn** [comment]
you will have to get the "real data" out of the "raw data" - there is good examples on how to do that: https://orchetect.github.io/OSCKit/documentation/osckit/osc-value-parsing

**Comment by orchetect** [comment]
> clearly, not in the right format

Yes, OSC messages are encoded and not meant to be converted to text directly. OSCKit will receive the messages and decode their contents so it's easy to use.

Check out the docs as @mk-conn suggests, and also check out the example projects that can help you get started: https://github.com/orchetect/OSCKit/tree/main/Examples
